src/main/java/seedu/gamebook/model/gameentry/Duration.java

```diff
@@ -78,30 +79,36 @@ public static boolean isValidDuration(String durationString) {
     public static boolean isValidDuration(Integer duration) {
         return duration != null
                 && (duration.equals(Integer.MIN_VALUE)
-                || duration >= 0);
+                || duration >= 0) && (duration < Integer.MAX_VALUE);
     }
 
+    /* added a try-catch block to prevent extremely large duration values from being entered, by handling the
+     NumberFormatException thrown by Integer::valueOf - Please feel free to edit if there is a better way */
```

Comment:
```suggestion
    /* added a try-catch block to prevent extremely large duration values from being entered, by handling the
     NumberFormatException thrown by Integer::valueOf */
```